src/uct/ib/dc/dc_mlx5.c

```diff
@@ -1388,7 +1369,24 @@ void uct_dc_mlx5_iface_restart_dci(uct_dc_mlx5_iface_t *iface,
     uct_rc_txqp_available_set(txqp, (int16_t)iface->super.super.config.tx_qp_len);
     uct_rc_txqp_purge_outstanding(&iface->super.super, txqp, ep_status,
                                   txwq->sw_pi, 0);
-    status = uct_dc_mlx5_iface_reset_dci(iface, &iface->tx.dcis[dci]);
+
+    ucs_assert(txwq->super.type == UCT_IB_MLX5_OBJ_TYPE_VERBS);
```

Comment:
is it possible to create DCI with DevX?
@Artemy-Mellanox

Reply:
Possible.

Reply:
hmmm, I copied this assert from original function
removed assert

Reply:
restored assert